arangod/RestHandler/RestAdminClusterHandler.cpp

```diff
@@ -0,0 +1,1563 @@
+      "database/collection not found");
+    return RestStatus::DONE;
+  }
+
+  ctx->collectionID = TRI_RidToString(collection->planId());
+  auto planPath = arangodb::cluster::paths::root()->arango()->plan();
+
+  VPackBuffer<uint8_t> trx;
+  {
+    VPackBuilder builder(trx);
+    arangodb::agency::envelope<VPackBuilder>::create(builder).read()
+      .key(planPath->dBServers()->str())
+      .key(planPath->collections()->database(ctx->database)
+        ->collection(ctx->collectionID)->str())
+      .done()
+    .done();
+  }
+
+  // gather information about that shard
+  return waitForFuture(AsyncAgencyComm().getValues(planPath).thenValue(
+      [this, ctx = std::move(ctx)](AgencyReadResult &&result) mutable {
+
+        if (result.ok()) {
+          switch (result.statusCode()) {
+            case fuerte::StatusOK:
+              return createMoveShard(std::move(ctx), result.slice().at(0));
+            case fuerte::StatusNotFound:
+              generateError(rest::ResponseCode::NOT_FOUND, TRI_ERROR_HTTP_NOT_FOUND, "unknown collection");
+              return futures::makeFuture();
+            default:
+              break;
+          }
+        }
+
+        // to some checks that this shard is moveable
+        generateError(result.asResult());
+        return futures::makeFuture();
+    }).thenError<VPackException>([this](VPackException const& e) {
+      generateError(Result{e.errorCode(), e.what()});
+    }).thenError<std::exception>([this](std::exception const& e) {
+      generateError(rest::ResponseCode::SERVER_ERROR, TRI_ERROR_HTTP_SERVER_ERROR, e.what());
+    }));
+}
+
+RestStatus RestAdminClusterHandler::handleQueryJobStatus() {
+  if (!ExecContext::current().isAdminUser()) {
+    generateError(rest::ResponseCode::FORBIDDEN, TRI_ERROR_HTTP_FORBIDDEN);
+    return RestStatus::DONE;
+  }
+
+  if (!ServerState::instance()->isCoordinator()) {
+    generateError(rest::ResponseCode::FORBIDDEN, TRI_ERROR_HTTP_FORBIDDEN, "only allowed on coordinators");
+    return RestStatus::DONE;
+  }
+
+  if (request()->requestType() != rest::RequestType::GET) {
+    generateError(rest::ResponseCode::METHOD_NOT_ALLOWED, TRI_ERROR_HTTP_METHOD_NOT_ALLOWED);
+    return RestStatus::DONE;
+  }
+
+  std::string jobId = request()->value("id");
+  if (jobId.empty()) {
+    generateError(rest::ResponseCode::BAD, TRI_ERROR_BAD_PARAMETER, "missing id parameter");
+    return RestStatus::DONE;
+  }
+
+  auto targetPath = arangodb::cluster::paths::root()->arango()->target();
+  std::vector<std::string> paths = {
+    targetPath->pending()->job(jobId)->str(),
+    targetPath->failed()->job(jobId)->str(),
+    targetPath->finished()->job(jobId)->str(),
+    targetPath->toDo()->job(jobId)->str(),
+  };
+
+  return waitForFuture(AsyncAgencyComm().sendTransaction(20s, AgencyReadTransaction{std::move(paths)})
+    .thenValue([this, targetPath, jobId = std::move(jobId)](AsyncAgencyCommResult &&result) {
```

Comment:
```suggestion
    .thenValue([this, targetPath, jobId = std::move(jobId)](AsyncAgencyCommResult&& result) {
```
do we need to capture `self` here?

Reply:
This lambda is executed (or destroyed) strictly before the `thenFinal` lambda in `waitForFuture` which captures `self`.